### PR TITLE
Use latest m-p-p and m-p-report-p also for the site of m-plugin-tools

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

**Comment by kwin** [comment]
@gnodet Any idea how to make this work?

**Comment by gnodet** [comment]
> @gnodet Any idea how to make this work?

I have this exception:
```
INFO] 
[INFO] --- maven-plugin-plugin:3.7.1-SNAPSHOT:helpmojo (generated-helpmojo) @ maven-plugin-plugin ---
[INFO] Downloading from apache.snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-plugin-plugin/3.7.1-SNAPSHOT/maven-plugin-plugin-3.7.1-20221026.201321-13.pom
[INFO] Downloaded from apache.snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-plugin-plugin/3.7.1-SNAPSHOT/maven-plugin-plugin-3.7.1-20221026.201321-13.pom (11 kB at 72 kB/s)
[WARN] The POM for org.apache.maven.plugins:maven-plugin-plugin:jar:3.7.1-20221026.201321-13 is invalid, transitive dependencies (if any) will not be available, enable verbose output (-X) for more details
[WARN] Error injecting: org.apache.maven.plugin.plugin.HelpGeneratorMojo
java.lang.NoClassDefFoundError: org/apache/maven/tools/plugin/generator/GeneratorException
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
```
I don't think we can have the plugin being packaged with itself.  Maven has to load the jar, but it has not been built yet.

File: pom.xml [hunk 1]
```diff
@@ -325,6 +325,22 @@
...
+            <externalJavadocBaseUrls>
+              <externalJavadocBaseUrl>https://docs.oracle.com/javase/8/docs/api/</externalJavadocBaseUrl>
+            </externalJavadocBaseUrls>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-plugin-report-plugin</artifactId>
+          <version>${project.version}</version>
+        </plugin>
```
Comment (slawekjaranowski) [inline pom.xml:343]: IMHO we can't use it now in this way ... there is egg and hen problem
We should use the latest released version

The `maven-plugin-report-plugin` has not been released yet

File: maven-plugin-plugin/pom.xml [hunk 2]
```diff
@@ -271,7 +271,7 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-plugin-plugin</artifactId>
+        <artifactId>maven-plugin-report-plugin</artifactId>
```
Comment (slawekjaranowski) [inline maven-plugin-plugin/pom.xml:274]: Here we will have a cycle ... `maven-plugin-report-plugin` also requires `maven-plugin-plugin`

**Comment by slawekjaranowski** [comment]
> So this may possible for the site which uses `m-plugin-report-p`, but not for `m-plugin-p` which uses... itself.

But we must do it in two steps .... first with `install` and second with `site` - I would like not to prefer do it in this way

**Comment by kwin** [comment]
> But we must do it in two steps .... first with `install` and second with `site` - I would like not to prefer do it in this way

Even if you do that it fails. Please try it locally. But if there is no easy way to achieve that, I am fine with sticking to the previous release.

**Pull request closed without merging**